anagram.cpp

```diff
@@ -1,13 +1,64 @@
 #include "anagram.h"
+using namespace std;
+
+void removeSpace(std::string& w) {
+    int count = 0;
+    for (int i = 0; w[i]; i++) {
+        if (w[i] != ' ') {
+            w[count] = w[i];
+            count++;
+        }
+    }
+    w[count] = '\0';
+}
+
+
+void string_align(std::string& w) {
+    std::transform(w.begin(), w.end(), w.begin(), ::tolower);
+    
+    //removeSpace(w);
+    
+    w.erase(remove(w.begin(), w.end(), ' '),w.end());
+}
+
+
+bool charWiseCompare(const std::string& w1, const std::string& w2) {
+    int size = w1.size();
+    for (int i = 0; i < size; i++) {
+        if (w1[i] != w2[i]) {
+            return false;
+        }
+    }
+    return true;
+}
 
 bool Anagram::WordPairIsAnagram(const std::string& word1, const std::string& word2) {
     //Fill the correct implementation here
-    return false;
```

Comment:
Extend the tests of this function with examples in the Wikipedia link, which is given in README.md. Especially the first three. Try to group the examples there, and test each group.

Reply:
From examples found a case with '-' involved, added it to the test cases.